imod/mf6/wel.py

```diff
@@ -494,18 +492,32 @@ def regrid_like(self, target_grid: GridDataArray, *_) -> "Well":
         grid-agnostic, instead this method clips the package based on the grid
         exterior.
         """
-        return clip_by_grid(self, target_grid)
+        target_grid_2d = target_grid.isel(layer=0, drop=True, missing_dims="ignore")
+        return clip_by_grid(self, target_grid_2d)
 
-    def mask(self, _) -> Package:
+    def mask(self, domain: GridDataArray) -> "Well":
         """
-        The Well package has no mask method implemented. Wells falling in
-        inactive cells are automatically removed in the call to write to
-        Modflow 6 package. You can verify this by calling the ``to_mf6_pkg``
-        method.
+        Mask wells based on two-dimensional domain. For three-dimensional
+        masking: Wells falling in inactive cells are automatically removed in
+        the call to write to Modflow 6 package. You can verify this by calling
+        the ``to_mf6_pkg`` method.
         """
-        # TODO: Add docsting message to logger
-        # message = textwrap.dedent(self.mask.__doc__)
-        return deepcopy(self)
+
+        # Drop layer coordinate if present, otherwise a layer coordinate is assigned
+        # which causes conflicts downstream when assigning wells and deriving
+        # cellids.
+        domain = domain.isel(layer=0, drop=True, missing_dims="ignore").drop(
+            "layer", errors="ignore"
+        )
+        point_active = points_values(domain, x=self.x, y=self.y)
+
+        is_inside_exterior = point_active == 1
+        selection = self.dataset.loc[{"index": is_inside_exterior}]
```

Comment:
The above lines is the same as the clip_by_grid method. Can't you call that method to reduce code duplication?